### ISSUE TITLE
Make twisted not relying on gc details (and run on pypy)

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3318]
|[<img alt="fijal's avatar" src="https://avatars.githubusercontent.com/u/85944?s=50" width="50" height="50">](https://github.com/fijal)| @fijal reported|
|-|-|
|Trac ID|trac#3318|
|Type|task|
|Created|2008-06-27 20:51:45Z|
|Branch|https://github.com/twisted/twisted/tree/pypy-compatibility-3318|

This is a general ticket for keeping track of pypy-twisted interaction.

<details><summary>Searchable metadata</summary>

```
trac-id__3318 3318
type__task task
reporter__fijal fijal
priority__normal normal
milestone__ 
branch__branches_pypy_compatibility_3318 branches/pypy-compatibility-3318
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1214599905000000 1214599905000000
changetime__1215519467000000 1215519467000000
version__None None
owner__fijal fijal

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="fijal's avatar" src="https://avatars.githubusercontent.com/u/85944?s=50" width="50" height="50">](https://github.com/fijal)<a name="note_1"></a>|@fijal commented|
|-|-|

(In [24116]) A branch to make twisted pypy-compatible (again)

**Comment by twisted-trac** [comment]
|[<img alt="fijal's avatar" src="https://avatars.githubusercontent.com/u/85944?s=50" width="50" height="50">](https://github.com/fijal)|@fijal removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @fijal|
|-|-|

twisted/web/sux.py and twisted/trial/test/test_tests.py have only white spaces change. Please revert these ones and merge, thanks!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [24240]) Merge pypy-compatibility-3318

Author: fijal
Reviewer: therve
Fixes: [#3318](https://github.com/twisted/twisted/issues/3318)

Remove a number of occurrences of the assumption that refcounting will cause
bytes written to a file without a flush or close to be available to be read
from that file immediately..  Also fix one occurrence of the assumption that
without an explicit call to gc.collect() a synchronously failing Deferred will
log its failure in a timely manner.
